blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
I am not sure I would immediately guess what's in `theano_expressions.py`.  It sounds for me like yet another abstraction level Blocks have on top of Theano. It is not clear that I should look there for `L2_norm` or other one-liners.

So I am fine with `theano_macros.py`, `theano_oneliners.py`, `oneliners.py`, but not with `theano_expressions.py`, because for me expression is not a callable object, and this name gives me much less clue on the content of the file compared to all aforementioned ones.